File: CuRoadmapBackend/Models/Course.cs
namespace CuRoadmapBackend.Models;

public sealed class Course
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? HandbookLink { get; set; }
    public CourseType CourseType { get; set; }
    public CourseCategory Category { get; set; }
    public int[]? AllowedCohorts { get; set; }
    public int[] AvailableSemesters { get; set; } = [];
    public int? RecommendedSemester { get; set; }
    public double Workload { get; set; }
    public double? CsatMetric { get; set; }
    public List<CourseDependency> CourseDependencies { get; set; } = [];
    public List<MajorRequirement> MajorRequirements { get; set; } = [];
    public List<Student> StudentsWhoPassed { get; set; } = [];
}

File: backend/store/interfaces/course.go
package interfaces

import (
	"github.com/cu-3rd-party/cu-roadmap/backend/domain/enums"
	"github.com/google/uuid"
)

type CourseData struct {
	ID                  uuid.UUID
	Title               string
	Description         *string
	HandbookLink        *string
	CourseType          enums.CourseType
	Category            enums.CourseCategory
	AllowedCohorts      []int
	AvailableSemesters  []int
	RecommendedSemester *int
	Workload            float64
	SeminarsWeek        int
	LecturesWeek        int
	AnalogGroup         string
	CsatMetric          *float64
	DisplayMode         enums.CourseDisplayMode
	Prerequisites       []uuid.UUID
	Corequisites        []uuid.UUID
	Postrequisites      []uuid.UUID
}

type CourseDependencyData struct {
	ID               uuid.UUID            `json:"id"`
	CourseID         uuid.UUID            `json:"course_id"`
	RequiredCourseID *uuid.UUID           `json:"required_course_id,omitempty"`
	RequiredGroupID  *uuid.UUID           `json:"required_group_id,omitempty"`
	DependencyType   enums.DependencyType `json:"dependency_type"`
	AlternativeGroup int                  `json:"alternative_group"`
}
